src/symmetric_heap_c.c

```diff
@@ -64,6 +64,9 @@
 #pragma weak shfree = pshfree
 #define shfree pshfree
 
+#pragma weak shmemx_malloc_with_hints = pshmem_malloc_with_hints
+#define shmemx_malloc_with_hints pshmem_malloc_with_hints
+
```

Comment:
@davidozog @rdesai16 It needs to be `pshmemx_...`, right?

Reply:
Yes, it should (at least, that would be consistent with what we've done in the past).

Great catch @wrrobin!

Reply:
yeah good point! will change that.